httpclient/httpclient.go

```diff
@@ -82,8 +82,9 @@ type Client struct {
 	tracer                tracer
 	noRateLimitBackoff    bool
 
-	mu      sync.RWMutex
-	last429 time.Time
+	mu                 sync.RWMutex
+	last429            time.Time
+	lastRetryAfterTime time.Time
```

Comment:
Rate limiting can happen on a per user/service/IP/endpoint bases. 

Is it wise for us to bucket that all together? Since the client side has little context to _how_ a rate limit enforced on the server side I'm not sure how much smarts we can build into a generic client.

Reply:
I think that's what we added the `NoRateLimitBackoff` option for, because we were getting user-based `429`s from `circle/circle` and rate limiting the whole client (which was shared across requests) leading to a noisy neighbours-like problem.